protocol/superc20/mirror-standard.md

```diff
@@ -0,0 +1,114 @@
+# Mirror Standard Design Doc
+
+## Purpose
+
+_The work present below is part of the [Interoperability project](https://github.com/ethereum-optimism/optimism/issues/10899)._
+
+This document presents the `SuperchainERC20` Mirror Standard, which enables existing tokens deployed through `OptimismMintableERC20` Factory in L2 (also called legacy tokens) to be interoperable without any liquidity migration process.
+
+## Summary
+
+We discussed different approaches to making existing liquidity interoperable in the [Migrated Liquidity Design Doc](https://github.com/ethereum-optimism/design-docs/pull/35). We introduced the Mirror standard and highlighted it as the simplest solution, as it requires no liquidity migration at all.
+
+The Mirror approach deploys a new token contract that follows the `ISuperchainERC20` specifications and will work as a double-entry point for an appointed `OptimismMintableERC20`. We must modify the `L2StandardBridge` to enable the Mirror contract to burn and mint the underlying `OptimismMintableERC20` to implement the Mirror.
+
+## Proposed Design
+
+The proposed design includes a specification on implementing a double-entry point to mirror an existing `OptimismMintableERC20`, as well as approaches to the Factory and `L2StandardBridge`.
+
+### OptimismMintableERC20 proxiable functions
+
+The following functions will be implemented as simple proxies, that is, by having a mirror call to the appointed `OptimismMintableERC20` token:
+
+- `totalSupply()`
+- `balanceOf(account)`
+- `transfer(recipient, amount)`
+- `transferFrom(sender, recipient, amount)`
+
+### Local functions
+
+Approvals functions will be implemented in a conventional way, specific to the contract, and not as mirror calls:
+
+- `approve(spender, amount)`
+- `allowance(owner, spender)`
+
+### Code sample
+
+Below, we show what the implementation of the mirror contract would look like.
+
+```solidity
+contract Mirror is Superc20 {
+  IERC20 public immutable LEGACY_TOKEN;
+  IL2StandardBridge public constant L2_STANDARD_BRIDGE = IL2StandardBridge(0x4200000000000000000000000000000000000010);
+
+  constructor(IERC20 _legacyToken) Superc20(_name, _symbol, _decimals) {
+    LEGACY_TOKEN = _legacyToken;
+    _name = sting.concat("Mirror",_legacyToken.name());
+    _symbol = string.concat("m:",_legacyToken.symbol());
+    _decimals = _legacyToken.decimals();
+  }
+
+  function totalSupply() public view override returns (uint256) {
+    return LEGACY_TOKEN.totalSupply();
+  }
+
+  function balanceOf(address account) public view override returns (uint256) {
+    return LEGACY_TOKEN.balanceOf(account);
+  }
+
+  function transfer(address recipient, uint256 amount) public override returns (bool) {
```

Comment:
It would be good to have the flow that shows a cross chain transfer, perhaps as a mermaid diagram. My current understanding, assuming the mirror token is already deployed on both domains:
- holding `OptimismMintableERC20Token`
- call mirror `transfer` to self
- call mirror superchain erc20 `sendERC20` per https://github.com/ethereum-optimism/specs/pull/71#discussion_r1631562710, this burns the token
- on remote domain, send the executing transaction that mints the token

Reply:
+1 I would suggest removing code from this design doc, and replace it with diagrams showing the user flows and data flows. Per the [design doc template](https://github.com/ethereum-optimism/design-docs/blob/main/assets/design-doc-template.md):

> As a rule of thumb, including code snippets (except for defining an external API) is likely too low level.

Reply:
I just created some basic flows in Figma: https://www.figma.com/board/IXC6WnDaqrheplxbCZpw3l/%5BExternal%5D-Mirror-Standard?node-id=0-1&t=3tsA67ghaqocKaRv-1

will migrate these to Mermaid after discussion